compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -6251,6 +6256,23 @@ private BLangType createAnonymousRecordType(RecordTypeDescriptorNode recordTypeD
         return createUserDefinedType(pos, (BLangIdentifier) TreeBuilder.createIdentifierNode(), typeDef.name);
     }
 
+    private BLangType createAnonymousTupleType(TupleTypeDescriptorNode tupleTypeDescriptorNode,
+                                                BLangTupleTypeNode tupleTypeNode) {
+        BLangTypeDefinition typeDef = (BLangTypeDefinition) TreeBuilder.createTypeDefinition();
+        Location pos = getPosition(tupleTypeDescriptorNode);
+        // Generate a name for the anonymous object
+        String genName = anonymousModelHelper.getNextAnonymousTypeKey(this.packageID, this.anonTypeNameSuffixes);
+        IdentifierNode anonTypeGenName = createIdentifier(symTable.builtinPos, genName);
+        typeDef.setName(anonTypeGenName);
+        typeDef.flagSet.add(Flag.PUBLIC);
+        typeDef.flagSet.add(Flag.ANONYMOUS);
+
+        typeDef.typeNode = tupleTypeNode;
+        typeDef.pos = pos;
+        addToTop(typeDef);
+        return createUserDefinedType(pos, (BLangIdentifier) TreeBuilder.createIdentifierNode(), typeDef.name);
+    }
```

Comment:
We don't need a separate method for this? We can refactor `createAnonymousRecordType` to work with tuples.

Reply:
Please recheck all methods. We should try and minimize duplicating logic.